guides/source/getting_started_with_devcontainer.md

```diff
@@ -0,0 +1,123 @@
+
+`rails-new`  generates a new Rails application for you without having to install Ruby on
+your machine. It uses Docker to generate the Rails application, thus allowing Docker to
+take care of installing the correct Ruby and Rails versions for you.
+
+To install rails-new, follow the installation instructions [in the readme](https://github.com/rails/rails-new?tab=readme-ov-file#installation).
+
+Creating the Blog Application
+-----------------------------
+
+Rails comes with a number of scripts called generators that are designed to make
+your development life easier by creating everything that's necessary to start
+working on a particular task. One of these is the new application generator,
+which will provide you with the foundation of a fresh Rails application so that
+you don't have to write it yourself. The `rails-new` tool uses this generator to
+create a new Rails application for you.
+
+NOTE: The examples below use `$` to represent your terminal prompt in a UNIX-like OS,
+though it may have been customized to appear differently.
+
+To use `rails-new` to generate your app, open a terminal, navigate to a directory where you have
+rights to create files, and run:
+
+```bash
+$ rails-new blog
+```
+
+This will create a Rails application called Blog in a `blog` directory.
+
+TIP: You can see all of the command line options that the Rails application
+generator accepts by running `rails-new --help`.
+
+After you create the blog application, switch to its folder:
+
+```bash
+$ cd blog
+```
+
+The `blog` directory will have a number of generated files and folders that make
+up the structure of a Rails application. Most of the work in this tutorial will
+happen in the `app` folder. For a full rundown of everything in your application
+see the full [Getting Started guide](getting_started.md#creating-the-blog-application).
+
+Opening the Blog Application in a Dev Container
+-----------------------------------------------
+
+Our new Rails application comes with a Dev Container already configured and ready to use.
+We will use VSCode to spin up and work with our Dev Container. Start by launching VSCode
+and opening your application.
+
+Once the application opens, VSCode should prompt you that a it has found a Dev Container
+configuration file, and you can reopen the folder in a Dev Container. Click the green "Reopen
+in Container" button to create the Dev Container.
+
+Once the Dev Container setup is complete, your development environment is ready to use,
+with Ruby, Rails, and all your dependencies installed.
+
+You can open the terminal within VScode to verify that Rails is installed:
+
+```bash
+$ rails --version
+Rails 7.2.0
+```
+
+You can now continue with the [Getting Started guide](getting_started.md#hello-rails) and
+begin building your Blog application. You will be working within VSCode, which serves as
+your entry point to your application's Dev Container, where you can run code, run tests and
+run your application.
```

Comment:
[This section](https://github.com/rails/rails/blob/4e3d2f7ea0ad5d939f82840c7ccc8db1a665eb1f/guides/source/getting_started.md#starting-up-the-web-server) in the [Getting Started guide](getting_started.md#hello-rails) talks about starting up the web server on your development machine, I'm wondering do we need a little more verbose instructions here about what happens once you get you able to open your blog application within Dev Container? Things like how do you run your server and see your application?

Reply:
Nope, everything is the exact same. If you `bin/rails s` within the container, it will boot the app and be accessible in your browser at `http://127.0.0.1:3000`.